### DISCUSSION TITLE
[libblkid] Why wholedisk_probe does not force GPT while gathering partition details?

### DISCUSSION THREAD

**Discussion by mfrydrysiak** [util-linux/util-linux#2146] [Q&A] [answered]
I patched my systemd-udevd to set `BLKID_PARTS_FORCE_GPT` flag on probing superblocks to recover from situations, when **Protective MBR is damaged**:

`blkid_probe_set_partitions_flags(pr, BLKID_PARTS_ENTRY_DETAILS | BLKID_PARTS_FORCE_GPT);`

It seems that this flag is indeed used while `partitions_probe` (in the `blkid_do_probe()` backend, once it detects the GPT layout). However, when it passes to details gathering (`details_only` label) and it executes `blkid_partitions_probe_partition` function, it opens a new disk prober in the `blkid_probe_get_wholedisk_probe` function, which does not contain the mentioned `BLKID_PARTS_FORCE_GPT` flag. As a result, as far as I understand, the protective MBR validation is not skipped (it fails) so it does not consider this partition layout as GPT.

Is there any way to avoid such situations so the pMBR validation is skipped?

util-linux version **v2.35.1**

**Comment by karelzak** [comment, marked as answer]
It seems like a bug; the flag should be copied to the whole-disk prober. I have fixed it in commit 22769acf56daf6683d635fe92e06ceb52e0433ff; it should be available in v2.39.

It would be nice if you could test the current git tree in your environment.

**Reply by mfrydrysiak** [reply]
Thanks, it did the trick :) I have applied and tested it on the v2.35.1 branch. Works as expected.

**Comment by karelzak** [comment]
And a suggestion, fix your PMBR :-)
